src/lib/classes/lsp3-universal-profile.spec.ts

```diff
@@ -181,7 +152,83 @@ describe('LSP3UniversalProfile', () => {
 
       const [result] = await universalProfile.connect(signers[0]).callStatic.getData([key]);
       const checkedsumResult = ethers.utils.getAddress(result);
-      expect(checkedsumResult).toEqual(secondController);
+      expect(checkedsumResult).toEqual(secondControllerAddress);
+    });
+
+    it.only('first account should be able to setData', async () => {
+      const abi = await universalProfile.populateTransaction.setData(
+        ['0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5'],
+        ['0xbeefbeef']
+      );
+
+      const result = await keyManager.connect(signers[0]).execute(abi.data);
+
+      expect(result).toBeTruthy();
+
+      const data = await universalProfile.getData([
+        '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+      ]);
+
+      expect(data).toEqual(['0xbeefbeef']);
+    });
+    it.only('second account should be able to setData', async () => {
+      const abi = await universalProfile.populateTransaction.setData(
+        ['0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5'],
+        ['0xcafecafe']
+      );
+
+      const result = await keyManager.connect(signers[1]).execute(abi.data);
+
+      expect(result).toBeTruthy();
+
+      const data = await universalProfile.getData([
+        '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+      ]);
+
+      expect(data).toEqual(['0xcafecafe']);
+    });
+  });
```

Comment:
![](https://memegenerator.net/img/instances/400x/81029058.jpg)

I think you can combine the 2 tests in one 

you can do something like:

[address1, address2].forEach((address) => it(YOUR TEST))

because the logic is the same.

Check this:
https://github.com/ERC725Alliance/erc725.js/blob/c3e352c525f5e2a6f72b6d41d708ff862a5248e3/src/lib/utils.test.ts#L171

https://blog.harveydelaney.com/running-a-test-with-multiple-test-cases-using-jest-enzyme-react/